src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -1045,13 +1044,59 @@ const bool ManchesterpatternDecoder::doDecode() {
 
 	while (i < pdec->messageLen)
 	{
-		
 		// Start vom MC Signal suchen
 		if (mc_start_found == false && (isLong(pdec->message[i]) || isShort(pdec->message[i])))
 		{
 			pdec->mstart = i;
 			mc_start_found = true;
 			mc_sync = true;
+
+			// lookup for first long
+			int pulseCnt = 0; 
+			bool preamble = false;
+			for (uint8_t l=i; l<pdec->messageLen; l++) {
+				bool pulseIsLong = isLong(pdec->message[l]);
+				
+				// no manchester
+				if (!(pulseIsLong || isShort(pdec->message[l]))) {
+					break;
+				}
+				
+				pulseCnt += (pulseIsLong ? 2 : 1);
+				// probe signal to match manchester
+				if (pulseIsLong) {
+					// probe clock based preamble
+					if (l == i && i > 0) {
+						int clock = (pdec->pattern[shorthigh] + abs(pdec->pattern[shortlow])) / 2;
+						int pClock = abs(pdec->pattern[pdec->message[l-1]]); 
+						int pClocks = round(pClock / (float)clock); 
```

Comment:
Fließkomma Berechnung sollte, sofern möglich, vermieden werden. 
Den Part verstehe ich auch noch nicht. Der Takt wird doch schon durch isManchester ermittelt.

Eine preamble ist mir in der Manchestercodierung auch nicht geläufig. Entweder das Signal ist Manchester codiert oder eben nicht.